python/foglamp/services/core/server.py

```diff
@@ -163,13 +165,28 @@ class Server:
             'displayName': 'Auth Providers',
             'order': '8'
         },
+        'authMethod': {
+            'description': 'Authentication method',
+            'type': 'enumeration',
+            'options': ["any", "password", "certificate"],
+            'default': 'any',
+            'displayName': 'Authentication method',
+            'order': '9'
+        },
         'certificateName': {
             'description': 'Certificate file name',
             'type': 'string',
             'default': 'foglamp',
             'displayName': 'Certificate Name',
             'order': '4'
         },
+        'authCertificateName': {
+            'description': 'CA Certificate file name',
+            'type': 'string',
+            'default': 'ca.cert',
```

Comment:
without extension it should be. And let's internally allow crt and cert both; later upload mechanism will allow .crt too and we should think of supporting .pem files.

the same logic we have for https